### PR TITLE
removing initmodule from ica

### PR DESCRIPTION
**Pull request by muku314115**
## Description

closes: #4639

### Commit Message / Changelog Entry

```text
remoing initmodule function from ica which mimicks initgenesis
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by DimitrisJim** [comment]
dope, thanks! Would you mind splitting this PR, one for each issue? This makes it easier and faster to review and merge.

File: modules/capability/module.go [hunk 1]
```diff
@@ -26,7 +26,6 @@ import (
 
 var (
 	_ module.AppModule           = (*AppModule)(nil)
-	_ module.AppModuleBasic      = (*AppModuleBasic)(nil)
```
Comment (DimitrisJim) [inline modules/capability/module.go:29]: note that theres associated functions that can additionally be removed, currently they should be duplicated (e.g `Name()` IIRC)

Comment (muku314115) [reply modules/capability/module.go]: Will incorporate this in another PR, for now edited the issues addressed and reverted changes

**Review by crodriguezvega — APPROVED** [review]
Thanks, @muku314115. This looks good to me. If you would like to add migration docs for v9 to explain what to do, you can take a look at the [v2-to-v3 migrations docs](https://github.com/cosmos/ibc-go/blob/main/docs/docs/05-migrations/04-v2-to-v3.md#upgrade-proposal) for reference. Otherwise we can add them ourselves later.

**Comment by muku314115** [comment]
> Thanks, @muku314115. This looks good to me. If you would like to add migration docs for v9 to explain what to do, you can take a look at the [v2-to-v3 migrations docs](https://github.com/cosmos/ibc-go/blob/main/docs/docs/05-migrations/04-v2-to-v3.md#upgrade-proposal) for reference. Otherwise we can add them ourselves later.

I am preparing another PR for some similar changes, will add the docs there as well

**Review by charleenfei — APPROVED** [review]
thanks @muku314115

**Review by damiannolan — APPROVED** [review]

**Pull request merged**